Dockerfile

```diff
@@ -0,0 +1,13 @@
+# syntax=docker/dockerfile:1
+
+FROM --platform=${BUILDPLATFORM} docker.io/tonistiigi/xx:golang AS xx
+FROM --platform=${BUILDPLATFORM} golang:alpine AS base
+COPY --from=xx / /
+WORKDIR /src
+
+FROM base AS tests
+ARG TARGETPLATFORM
+RUN --mount=target=. \
+    --mount=type=cache,target=/go/pkg \
+    --mount=type=cache,target=/root/.cache \
+      go test -tags 'netgo osusergo static_build' -v -cover ./...
```

Comment:
I like to use `-covermode=count -coverprofile=cover.out` to later generate a more informative coverage report with `go tool cover -html=cover.out -o cover.html`. But I think this is OK for now, I can look into how to improve these things one I am more familiar with our CI setup.